app/grandchallenge/challenges/static/js/challenges/dropdown.js

```diff
@@ -1,51 +1,79 @@
 
 function handle_nav_tab_dropdown() {
-    if (window.innerWidth < 992) {
+    if (window.innerWidth < 576) {
+        // 576 px is the bootstrap breakpoint for sm
+        navs_to_dropdown(false, true);
+    } else if (window.innerWidth < 992) {
         // 992 px is the bootstrap breakpoint for lg
         navs_to_dropdown();
     } else {
         navs_to_dropdown(true);
-    }
+        }
 }
 
-function navs_to_dropdown(reverse = false) {
+function navs_to_dropdown(reverse = false, horiz_page_pills = false) {
     // Get all elements that need to change class
-    const containers = document.getElementsByClassName("nav-tab-dropdown-container");
+    const containers = document.querySelectorAll('.nav-tab-dropdown-container,.nav-pill-dropdown-container');
+    const page_containers = document.getElementsByClassName("nav-pill-pages-container");
 
     const ulDropdownClasses = ["dropdown-menu", "dropdown-menu-left"];
     const ulTabClasses = ["nav", "nav-tabs", "border-0"];
     const itemDropdownClasses = ["dropdown-item", "challengeDropdown"];
-    const itemTabClasses = ["nav-item"];
+    const itemTabPillClasses = ["nav-item"];
+    const ulPillClasses = ["nav", "nav-pills", "col-12", "mb-3"];
+    const ulHorizPagePillClasses = ["d-flex", "align-items-stretch", "mb-3"];
+    const ulVertPagePillClasses = ["flex-column"];
+
+    for (let page_container of page_containers) {
+        let ul_pages  = page_container.querySelector("ul");
```

Comment:
It's a bit counterintuitive, but both of these variables can be `const` instead of `let` since they are both not reassigned inside the loop. They are changed every iteration, but when using const there you redefine (!=reassign) them. Using `const` wherever possible is preferred because this provides some protection by explicitly not allowing the variables to be reassigned.